src/gui/sharedialog.cpp

```diff
@@ -150,13 +150,19 @@ ShareDialog::ShareDialog(QPointer<AccountState> accountState,
         connect(_manager, &ShareManager::linkShareCreated, this, &ShareDialog::slotAddLinkShareWidget);
         connect(_manager, &ShareManager::linkShareRequiresPassword, this, &ShareDialog::slotLinkShareRequiresPassword);
     }
+    
+    _scrollAreaViewPort = new QWidget(_ui->scrollArea);
+    _scrollAreaLayout = new QVBoxLayout(_scrollAreaViewPort);
+    _scrollAreaLayout->setContentsMargins(0, 0, 0, 0);
+    _ui->scrollArea->setWidget(_scrollAreaViewPort);
 }
 
 ShareLinkWidget *ShareDialog::addLinkShareWidget(const QSharedPointer<LinkShare> &linkShare)
 {
-    _linkWidgetList.append(new ShareLinkWidget(_accountState->account(), _sharePath, _localPath, _maxSharingPermissions, this));
-
-    const auto linkShareWidget = _linkWidgetList.at(_linkWidgetList.size() - 1);
+    _linkWidgetList.append(new ShareLinkWidget(_accountState->account(), _sharePath, _localPath, _maxSharingPermissions, _ui->scrollArea));
+    
+    const auto index = _linkWidgetList.size() - 1;
+    const auto linkShareWidget = _linkWidgetList.at(index);
```

Comment:
@camilasan Do you need that `index` variable? Also, won't this work similarly `const auto linkShareWidget = _linkWidgetList.last();` ?

Reply:
leftover from something I was trying before but now I rewrote it better (imho) and I don't need last() or at().